src/Catalyst.Core.Modules.Keystore.Tests/IntegrationTests/KeyApiTest.cs

```diff
@@ -0,0 +1,351 @@
+* (at your option) any later version.
+*
+* Catalyst.Node is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Catalyst.Node. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#endregion
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Security;
+using System.Threading.Tasks;
+using Catalyst.Abstractions.Keystore;
+using Catalyst.Abstractions.Options;
+using Catalyst.Core.Lib.Config;
+using Catalyst.TestUtils;
+using Lib.P2P;
+using NSubstitute;
+using Org.BouncyCastle.Crypto.Parameters;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Catalyst.Core.Modules.Keystore.Tests.IntegrationTests
+{
+    public sealed class KeyApiTest : FileSystemBasedTest
+    {
+        private readonly IKeyStoreService _keyStoreService;
+
+        public KeyApiTest(ITestOutputHelper output) : base(output)
+        {
+            var dfsOptions = new DfsOptions(new BlockOptions(), new DiscoveryOptions(), new RepositoryOptions(FileSystem, Constants.DfsDataSubDir));
+            _keyStoreService = new KeyStoreService(dfsOptions);
+            _keyStoreService.SetPassphraseAsync(new SecureString()).Wait();
+        }
+
+        [Fact]
+        public async Task Self_Key_Exists()
+        {
+            const string name = "self";
+            var key = await _keyStoreService.CreateAsync(name, "ed25519", 0);
+
+            var keys = await _keyStoreService.ListAsync();
+            var self = keys.Single(k => k.Name == "self");
+
+            var me = await _keyStoreService.FindKeyByNameAsync("self");
+            Assert.Equal("self", self.Name);
+            Assert.Equal(me.Id, self.Id);
+        }
+
+        [Fact]
+        public async Task Export_Import()
+        {
+            const string name = "self";
+            var key = await _keyStoreService.CreateAsync(name, "ed25519", 0);
+            var password = "password".ToCharArray();
+            var pem = await _keyStoreService.ExportAsync(name, password);
+            Assert.StartsWith("-----BEGIN ENCRYPTED PRIVATE KEY-----", pem);
+
+            var keys = await _keyStoreService.ListAsync();
+            var self = keys.Single(k => k.Name == name);
+
+            await _keyStoreService.RemoveAsync("clone");
+            var clone = await _keyStoreService.ImportAsync("clone", pem, password);
+            Assert.Equal("clone", clone.Name);
+            Assert.Equal(self.Id, clone.Id);
+        }
+
+        [Fact]
+        public void Export_Unknown_Key()
+        {
+            var password = "password".ToCharArray();
+            ExceptionAssert.Throws<Exception>(() =>
+            {
+                var x = _keyStoreService.ExportAsync("unknown", password).Result;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unused local variable 'x'.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)